### PR TITLE
User can find hotels

### PR DESCRIPTION
**Pull request by EevanR**
[PT](https://www.pivotaltracker.com/story/show/171054022)

USER STORY

As a user
In order to have accommodation
I want to be provided with hotel recommendations

- add hotel controller with create action
- add hotels model
- make association to trip model
- find 3 top rated hotels closed to triangulated center of activities positions

### CODE REVIEW

File: app/controllers/api/v1/hotels_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          lng: lng,
+          price: price,
+          address: address,
+          trip_id: params[:trip]
+        )
+        hotels << hotel if hotel.persisted?
+      end
+      hotels
+    end
+    binding.pry
```
Comment (mantanquzzmo) [inline app/controllers/api/v1/hotels_controller.rb]: please remove

File: app/controllers/api/v1/trips_controller.rb [hunk 2]
```diff
@@ -8,6 +8,8 @@ def create
                        lng: params[:lng],
                        days: params[:days])
 
+
```
Comment (mantanquzzmo) [inline app/controllers/api/v1/trips_controller.rb:11]: excessive

File: app/controllers/api/v1/hotels_controller.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      longs = []
+      activity_type.activities.each do |i|
+        lat = i['lat']
+        lng = i['lng']
+        
+        lats << lat.to_f
+        longs << lng.to_f
+      end
+      avg_lat = lats.inject{ |sum, el| sum + el } / lats.size
+      avg_lng = longs.inject{ |sum, el| sum + el } / longs.size
```
Comment (sundqvistjohan) [inline app/controllers/api/v1/hotels_controller.rb]: it appears Ruby 2.4+ introduced `Array.sum`, so you could use
`avg_lat = lats.sum / lats.size`

File: app/controllers/api/v1/hotels_controller.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    
+    if activity_type.activities != []
+      lats = []
+      longs = []
+      activity_type.activities.each do |i|
+        lat = i['lat']
+        lng = i['lng']
+        
+        lats << lat.to_f
+        longs << lng.to_f
```
Comment (sundqvistjohan) [inline app/controllers/api/v1/hotels_controller.rb]: `i` is not a great variable name, `activity`? But not a blocker. Could also be shortened to below but up to you
```
lats << activity['lat'].to_f
longs << activity['lng'].to_f
```

**Review by sundqvistjohan — COMMENTED** [review]
Also remove fixture and stub if not using it

File: app/controllers/api/v1/hotels_controller.rb (hunk 1 above)
Comment (EevanR) [reply app/controllers/api/v1/hotels_controller.rb]: done

File: app/controllers/api/v1/hotels_controller.rb (hunk 3 above)
Comment (EevanR) [reply app/controllers/api/v1/hotels_controller.rb]: yup cool, changed

Comment (EevanR) [reply app/controllers/api/v1/hotels_controller.rb]: changed to 'hotel'

File: app/controllers/api/v1/hotels_controller.rb (hunk 4 above)
Comment (EevanR) [reply app/controllers/api/v1/hotels_controller.rb]: changed to activities, and refactored the push into lats, longs arrays

**Review by sundqvistjohan — COMMENTED** [review]
Remove simplecov if not using it

**Review by sundqvistjohan — APPROVED** [review]

**Pull request merged**